File: backend/app/schemas/ai.py
"""AI 助手 Schema（Stage 8，对齐 PRD §5.8）。"""

import uuid
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class AIConversationCreate(BaseModel):
    """创建 AI 会话请求。"""

    mode: str = Field("general", description="会话模式：trade_analysis/strategy/book_qa/risk_diagnosis/general")
    title: Optional[str] = None
    context: Optional[Dict[str, Any]] = None


class AIConversationUpdate(BaseModel):
    """更新 AI 会话请求（重命名）。"""

    title: Optional[str] = Field(None, max_length=200)


class AIConversationResponse(BaseModel):
    """AI 会话响应。"""

    id: uuid.UUID
    user_id: uuid.UUID
    mode: str
    title: Optional[str] = None
    context: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AIMessageCreate(BaseModel):
    """发送 AI 消息请求。"""

    content: str = Field(..., description="消息内容")
    context: Optional[Dict[str, Any]] = None


class AIMessageResponse(BaseModel):
    """AI 消息响应。"""

    id: uuid.UUID
    conversation_id: uuid.UUID
    role: str
    content: str
    tokens_used: Optional[int] = None
    feedback: str = "none"
    created_at: datetime

    model_config = {"from_attributes": True}


class AIMessageFeedback(BaseModel):
    """消息反馈请求。"""

    feedback: str = Field(..., description="反馈：like / dislike / none")


class AIChatResponse(BaseModel):
    """AI 聊天完整响应（包含用户消息和 AI 回复）。"""

    user_message: AIMessageResponse
    assistant_message: AIMessageResponse


# ---------- 信号 ----------


class AISignalRequest(BaseModel):
    """AI 生成交易信号请求。"""

    symbol: str = Field(..., description="交易对")
    strategy_id: Optional[uuid.UUID] = None
    context: Optional[Dict[str, Any]] = None


class AISignalResponse(BaseModel):
    """AI 生成交易信号响应。"""

    id: uuid.UUID
    symbol: str
    side: str  # buy / sell / hold
    strength: float  # 0.0 ~ 1.0
    reason: str
    source: str = "ai"
    status: str = "pending"
    strategy_id: Optional[uuid.UUID] = None
    context: Optional[Dict[str, Any]] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class SignalMarkRequest(BaseModel):
    """信号标记请求。"""

    status: str = Field(..., description="标记状态：adopted / ignored / executed")


# ---------- 报告 ----------


class AIReportRequest(BaseModel):
    """AI 生成分析报告请求。"""

    report_type: str = Field("trade", description="报告类型：trade/strategy/portfolio")
    period: str = Field("custom", description="报告周期：daily/weekly/monthly/custom")
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    context: Optional[Dict[str, Any]] = None


class AIReportResponse(BaseModel):
    """AI 生成分析报告响应。"""

    id: uuid.UUID
    report_type: str
    period: str
    title: str
    content: str
    summary: Optional[str] = None
    period_start: Optional[datetime] = None
    period_end: Optional[datetime] = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/ai.ts
// AI 助手模式
export type AiMode = 'general' | 'market' | 'strategy' | 'risk' | 'tutor';

// AI 会话
export interface AiConversation {
  id: string;
  user_id: string;
  title: string | null;
  mode: AiMode;
  message_count?: number;
  last_message_at?: string;
  context?: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

// AI 消息角色
export type AiMessageRole = 'user' | 'assistant' | 'system';

// AI 消息
export interface AiMessage {
  id: string;
  conversation_id: string;
  role: AiMessageRole;
  content: string;
  model?: string;
  tokens?: number;
  tokens_used?: number;
  created_at: string;
}

// AI 聊天响应（用户消息 + AI 回复）
export interface AiChatResponse {
  user_message: AiMessage;
  assistant_message: AiMessage;
}

// 创建会话请求
export interface ConversationCreateData {
  title?: string;
  mode: AiMode;
  context?: Record<string, unknown>;
}

// 发送消息请求
export interface ChatRequest {
  conversation_id?: string;
  message: string;
  mode?: AiMode;
  model?: string;
  stream?: boolean;
  context?: Record<string, unknown>;
}

// 交易信号方向
export type TradingSignalSide = 'buy' | 'sell' | 'hold';

// 交易信号
export interface TradingSignal {
  symbol: string;
  side: TradingSignalSide;
  entry_price?: number;
  stop_loss?: number;
  take_profit?: number;
  confidence?: number;
  strength?: number;
  reason: string;
  timeframe?: string;
  strategy_id?: string;
  created_at?: string;
}

// 报告周期
export type ReportPeriod = 'daily' | 'weekly' | 'monthly';

// AI 分析报告
export interface AiReport {
  id?: string;
  title: string;
  period?: ReportPeriod;
  report_type?: string;
  summary?: string;
  content?: string;
  key_points?: string[];
  generated_at: string;
}

// 交易信号请求
export interface SignalRequest {
  symbols: string[];
  symbol?: string;
  strategy_type?: string;
  strategy_id?: string;
  timeframe?: string;
  context?: Record<string, unknown>;
}

// 报告参数
export interface ReportParams {
  period: ReportPeriod;
  report_type?: string;
  start_date?: string;
  end_date?: string;
  context?: Record<string, unknown>;
}
